tracing-subscriber/src/registry/mod.rs

```diff
@@ -171,29 +173,102 @@ pub struct SpanRef<'a, R: LookupSpan<'a>> {
     data: R::Data,
 }
 
-/// An iterator over the parents of a span.
-///
-/// This is returned by the [`SpanRef::parents`] method.
+/// An iterator over the parents of a span, ordered from leaf to root.
 ///
-/// [`SpanRef::parents`]: struct.SpanRef.html#method.parents
+/// This is returned by the [`SpanRef::scope`] method.
 #[derive(Debug)]
-pub struct Parents<'a, R> {
+pub struct Scope<'a, R> {
     registry: &'a R,
     next: Option<Id>,
 }
 
+impl<'a, R> Scope<'a, R>
+where
+    R: LookupSpan<'a>,
+{
+    /// Flips the order of the iterator, so that it is ordered from root to leaf.
+
+    /// **Note**: if the "smallvec" feature flag is not enabled, this may allocate.
```

Comment:
nit: it _may_ allocate even if the "smallvec" feature is enabled, if we go over the size of the `SmallVec`. it will _always_ allocate if the feature is not enabled.

Reply:
Good catch. In my defence, this was copy/pasted from `SpanRef::from_root`. Oops.